### PR TITLE
DietPi-Dashboard | Enable password protection with global software password by default

### PR DESCRIPTION
**Pull request by nonnorm**
**Status**: Ready
- [x] 0.3.2 dashboard release

**Reference**: https://github.com/ravenclaw900/DietPi-Dashboard/issues/11

**Commit list/description**:

- DietPi-Software | Add password protection to config file

### CODE REVIEW

**Comment by MichaIng** [comment]
What do you think about maintaining a default configuration on the dashboard repository and downloading it on install? That way it is guaranteed that it contains the latest settings, also based on nightly vs stable branch.

Instead of `dietpi`, it should use `$GLOBAL_PW`.

Another idea I have is to by default comment settings which match the default anyway, so they are present for documentation reasons. Generally makes sense to have the file mostly self-explaining so that one does not necessarily need to read docs elsewhere.

**Comment by ravenclaw900** [comment]
That sounds good, however the config would probably have to be put in with the release assets, as I use the `main` branch for everything, and tags for managing releases. Commenting out those settings would be fine, since the dashboard puts in the defaults automatically anyway.

**Comment by MichaIng** [comment]
> That sounds good, however the config would probably have to be put in with the release assets

Ah yes that would be even cleaner, one download less. Having it somewhere in the repo also serves as kinda self-documentation for repo visitors who want to setup themselves 🙂.

**Comment by MichaIng** [comment]
Check out my comment, what do you think about that, adding also the default value as separate comment, so that it is preserved even when users change it?

**Comment by ravenclaw900** [comment]
Looks good, I'll add it to the dashboard repo.

File: dietpi/dietpi-software [hunk 1]
```diff
@@ -3526,31 +3526,20 @@ _EOF_
...
-
-# Password protection
-# - Default: false
-pass = true
-hash = "$(echo -n "$GLOBAL_PW" | sha512sum | mawk '{print $1}')"
-secret = "$(openssl rand -hex 32)"
-_EOF_
+			G_CONFIG_INJECT '#pass[[:blank:]]=' 'pass = true' /opt/dietpi-dashboard/config.toml
+			G_CONFIG_INJECT '#hash[[:blank:]]=' "hash = \"$(echo -n "$GLOBAL_PW" | sha512sum | mawk '{print $1}')\"" /opt/dietpi-dashboard/config.toml
+			G_CONFIG_INJECT '#secret[[:blank:]]=' "secret = \"$(openssl rand -hex 32)\"" /opt/dietpi-dashboard/config.toml
```
Comment (MichaIng) [inline dietpi/dietpi-software]: We must not reset the password and the secret on a reinstall. What about defining the URL for the config based on the choice and then do the download and config changes in a separate `[[ ! -f '/opt/dietpi-dashboard/config.toml' ]]` block below?

**Review by MichaIng — APPROVED** [review]

**Comment by MichaIng** [comment]
We need to add a tab to our documentation about how to add password protection (when installed before this release), how to reset the password and how to forcefully invalidate session tokens by changing the secret: https://github.com/MichaIng/DietPi-Docs/issues/620

**Pull request merged**